### ISSUE TITLE
Building click in ubuntu 18.04

### ISSUE THREAD

**Issue by fruganyumisa** [Wi5/odin#56]
I adopted the firmware built for netgear r6100 but when I try using the click and click-align binaries dont work they give an error, Am I supposed to use my own click and click-align applications in the APs. If yes what environment do i need to build them I use ubuntu 18.04

**Comment by josemariasaldana** [comment]
Hi, you also need to cross-complie click and click-align binaries. They have to work in the Netgear, so you cannot use those compiled for Ubuntu.

Follow these instructions: https://github.com/Wi5/odin-wi5/wiki/Cross-compiling-Click-Modular-Router-for-Odin

**Comment by fruganyumisa** [comment]
Thank you @josemariasaldana but I am a bit confused when you say I have to use binaries that work in the Netgear, I want to build click on ubuntu 18.04 and ship it to Netgear is that correct ? Assuming that I created first the environment to build openwrt firmware for Netgear

**Comment by josemariasaldana** [comment]
Yes, that is correct. You first build in your Ubuntu machine an environment for building OpenWrt
https://github.com/Wi5/odin-wi5/wiki/Create-OpenWrt-bin-image-including-openvswitch-and-the-ath9k-patch
https://oldwiki.archive.openwrt.org/toh/netgear/netgear_r6100

And then you use that same environment for cross-compiling Click (cross-compiling means that you compile in a machine an executable file that will run in another machine)
https://github.com/Wi5/odin-wi5/wiki/Cross-compiling-Click-Modular-Router-for-Odin

And then you copy the cross-compiled executable from Ubuntu to the Netgear.
